src/engines/aurora/walkeleveval.h

```diff
@@ -0,0 +1,45 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Class to evaluate walkmesh elevation at given coordinates. Code ported from
+ *  http://geomalgorithms.com/a06-_intersect-2.html
+ */
+
+#ifndef ENGINES_AURORA_WALKELEVEVAL_H
+#define ENGINES_AURORA_WALKELEVEVAL_H
+
+#include "src/engines/aurora/walkmesh.h"
+
+namespace Engines {
+
+class WalkmeshElevationEvaluator {
+public:
+	void setWalkmesh(Walkmesh *w);
+
+	/** Return elevation at given coordinates or FLT_MIN if can't walk there. */
+	float getElevationAt(float x, float y) const;
```

Comment:
If you want to keep this as a separate class, at least make it a utility class without state. Have a `static float getElevationAt(Walkmesh &w, float x, float y)` method.